### PR TITLE
Added xmlrpc support to the nrpe package

### PR DESCRIPTION
**Pull request by infraweavers**
Will only work on Pfsense > 2.4 however as this is adding a feature, we think it's reasonable to not have backwards compatibility here.

### CODE REVIEW

**Comment by infraweavers** [comment]
@rbgarga All done here too

**Review by rbgarga — APPROVED** [review]

File: net-mgmt/pfSense-pkg-nrpe/files/usr/local/pkg/nrpe.inc [hunk 1]
```diff
@@ -253,4 +254,69 @@ function nrpe_custom_php_validation_command($post, &$input_errors) {
...
+	/* Now assemble a command to execute the         */
+	/* previously sent PHP file in the background.   */
+	/*************************************************/
+	$execcmd2 = "exec(\"/usr/local/bin/php-cgi -f '/tmp/remote_nrpe_config_apply.php' > /dev/null 2>&1 &\");";
+	$synctimeout = intval($synctimeout);
+	$rpc_client = new pfsense_xmlrpc_client();
+	$rpc_client->setConnectionData($sync_to_ip, $port, $username, $password, $protocol);
+	$resp = $rpc_client->xmlrpc_method('merge_installedpackages_section', $xml, $synctimeout);
+	$resp = $rpc_client->xmlrpc_exec_php($execcmd, $synctimeout);
+	$resp = $rpc_client->xmlrpc_exec_php($execcmd2, $synctimeout);
```
Comment (jim-p) [inline net-mgmt/pfSense-pkg-nrpe/files/usr/local/pkg/nrpe.inc:319]: Same concern here as with sudo. This looks unnecessary when the PHP can be run directly.

**Review by jim-p — CHANGES_REQUESTED** [review]

File: net-mgmt/pfSense-pkg-nrpe/files/usr/local/pkg/nrpe.inc (hunk 1 above)
Comment (infraweavers) [reply net-mgmt/pfSense-pkg-nrpe/files/usr/local/pkg/nrpe.inc]: Changed

**Review by jim-p — APPROVED** [review]

**Comment by rbgarga** [comment]
Manually merged to get commits squashed.  Thanks!

**Pull request closed without merging**